streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -109,16 +109,27 @@ public StateDirectory(final StreamsConfig config, final Time time, final boolean
                 log.warn("Using /tmp directory in the state.dir property can cause failures with writing the checkpoint file" +
                     " due to the fact that this directory can be cleared by the OS");
             }
-
             // change the dir permission to "rwxr-x---" to avoid world readable
-            final Path basePath = Paths.get(baseDir.getPath());
-            final Path statePath = Paths.get(stateDir.getPath());
+            configurePermissions(Paths.get(baseDir.getPath()));
+            configurePermissions(Paths.get(stateDir.getPath()));
+        }
+    }
+    
+    private void configurePermissions(final Path path) {
+        if (path.getFileSystem().supportedFileAttributeViews().contains("posix")) {
             final Set<PosixFilePermission> perms = PosixFilePermissions.fromString("rwxr-x---");
             try {
-                Files.setPosixFilePermissions(basePath, perms);
-                Files.setPosixFilePermissions(statePath, perms);
+                Files.setPosixFilePermissions(path, perms);
             } catch (final IOException e) {
-                log.error("Error changing permissions for the state or base directory {} ", stateDir.getPath(), e);
+                log.error("Error changing permissions for the directory {} ", path, e);
+            }
+        } else {
+            final File file = path.toFile();
+            boolean set = file.setReadable(true, false);
+            set &= file.setWritable(true, true);
```

Comment:
If you only have `file.setWritable(true, true)` then the directory will still be writeable by non-users, I assume?  I actually don't know the details of the `File#setXXX` methods -- but we don't want it to be writeable by just anyone. Should we instead do something like

```suggestion
            set &= file.setWritable(false) && file.setWritable(true, true);
```

Reply:
The `File` API only allows you to set permissions for the owner or for everyone. A call to `setWritable(writable)` just delegates to `setWritable(writable, true)`. I prefer to call `setWritable(writable, ownerOnly)` directly as I think the intent is slightly clearer. Calling `setWritable(true, true)` or `setWritable(true)` makes the file writable only by the owner.

Reply:
Ah, I see. Thanks for the explanation